prospero-metadata/src/main/java/org/wildfly/prospero/metadata/ManifestVersionResolver.java

```diff
@@ -89,19 +93,53 @@ public ManifestVersionRecord getCurrentVersions(List<Channel> channels) throws I
             } else if (manifestCoordinate.getUrl() != null) {
                 final String content = read(manifestCoordinate.getUrl());
                 final String hashCode = HashUtils.hash(content);
-                manifestVersionRecord.addManifest(new ManifestVersionRecord.UrlManifest(manifestCoordinate.getUrl().toExternalForm(), hashCode));
+                final ChannelManifest manifest = ChannelManifestMapper.from(manifestCoordinate.getUrl());
+                manifestVersionRecord.addManifest(new ManifestVersionRecord.UrlManifest(manifestCoordinate.getUrl().toExternalForm(), hashCode, manifest.getName()));
             } else if (manifestCoordinate.getVersion() != null) {
-                manifestVersionRecord.addManifest(new ManifestVersionRecord.MavenManifest(manifestCoordinate.getGroupId(), manifestCoordinate.getArtifactId(), manifestCoordinate.getVersion()));
+                final String description = getManifestDescription(manifestCoordinate, resolverFactory.create(channel.getRepositories()));
+                manifestVersionRecord.addManifest(new ManifestVersionRecord.MavenManifest(
+                        manifestCoordinate.getGroupId(),
+                        manifestCoordinate.getArtifactId(),
+                        manifestCoordinate.getVersion(),
+                        description));
             } else {
                 final MavenVersionsResolver mavenVersionsResolver = resolverFactory.create(channel.getRepositories());
                 final Optional<String> latestVersion = VersionMatcher.getLatestVersion(mavenVersionsResolver.getAllVersions(manifestCoordinate.getGroupId(), manifestCoordinate.getArtifactId(),
                         manifestCoordinate.getExtension(), manifestCoordinate.getClassifier()));
-                manifestVersionRecord.addManifest(new ManifestVersionRecord.MavenManifest(manifestCoordinate.getGroupId(), manifestCoordinate.getArtifactId(), latestVersion.get()));
+                if (latestVersion.isPresent()) {
+                    final String description = getManifestDescription(new ChannelManifestCoordinate(
+                            manifestCoordinate.getGroupId(), manifestCoordinate.getArtifactId(), latestVersion.get()
+                    ), mavenVersionsResolver);
+                    manifestVersionRecord.addManifest(new ManifestVersionRecord.MavenManifest(
+                            manifestCoordinate.getGroupId(),
+                            manifestCoordinate.getArtifactId(),
+                            latestVersion.get(),
+                            description));
+                } else {
```

Comment:
In this case the version is resolved from an offline repository which returns it from already resolved maven cache producing the last used version of the manifest.

Yes the record will show the version of the resolved manifest

Reply:
That's good point, yes in case of a revert we cannot rely on the version in repository. When reverting a server, the [version record from the previous state](https://github.com/wildfly-extras/prospero/blob/abad88839eec55573f62980ea3c260b4a14cd657/prospero-common/src/main/java/org/wildfly/prospero/actions/InstallationHistoryAction.java#L130) is used instead.